File: app/src/main/java/pl/futuredev/capstoneproject/viewmodel/CityViewModel.java
package pl.futuredev.capstoneproject.viewmodel;

import android.arch.lifecycle.LiveData;
import android.arch.lifecycle.ViewModel;

import pl.futuredev.capstoneproject.database.entity.CityPOJO;
import pl.futuredev.capstoneproject.database.repository.CityItemRepository;

public class CityViewModel extends ViewModel {

    private CityItemRepository cityItemRepository;

    public CityViewModel(CityItemRepository cityItemRepository) {
        this.cityItemRepository = cityItemRepository;
    }

    public LiveData<CityPOJO> getCityById(String cityId) {
        return cityItemRepository.getCityById(cityId);
    }

}

File: app/src/main/java/pl/futuredev/capstoneproject/ui/viewmodels/CityViewModel.kt
package pl.futuredev.capstoneproject.ui.viewmodels

import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.CoroutineDispatcher
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import pl.futuredev.capstoneproject.data.local.entities.City
import pl.futuredev.capstoneproject.data.local.repositories.CityRepository

class CityViewModel @ViewModelInject constructor(
    private val repository: CityRepository,
    private val dispatcher: CoroutineDispatcher = Dispatchers.Main
) : ViewModel() {

    private var isLiked: LiveData<City>? = null

    fun isLikedCity(name: String): LiveData<City>? {
        if (isLiked == null) {
            isLiked = repository.getCityByName(name)
        }
        return isLiked
    }

    fun likeCity(city: City) {
        viewModelScope.launch(dispatcher) {
            repository.insertCity(city)
        }
    }

    fun removeFromLikedCity(name: String) {
        viewModelScope.launch(dispatcher) {
            repository.deleteCityByName(name)
        }
    }

}